Update copyright headers

bernhardmgruber opened the pull request:
Since I am sloppy with updating the copyright in file headers I bulk updated them now using the following command:

~~`git log --no-merges --author="Bernhard Manfred Gruber"  --name-only --pretty=format:"" | sort -u | xargs sed -i -e '/Bernhard Manfred Gruber/ s/Copyright [[:digit:]-]\+/Copyright 2022/ ; /Bernhard Manfred Gruber/! s/Copyright [[:digit:]-]\+ \(.*\)/Copyright 2022 \1, Bernhard Manfred Gruber/'`~~

~~It's not perfect because it sets the year 2022 everywhere.~~

Commands:
`git log --no-merges --after="2020-01-01 00:00" --before="2021-01-01 00:00" --author="Bernhard Manfred Gruber" --name-only --pretty=format:""  | sort -u  | xargs sed -i -e '/Bernhard Manfred Gruber/ s/Copyright [[:digit:]-]\+/Copyright 2020/ ; /Bernhard Manfred Gruber/! s/Copyright [[:digit:]-]\+ \(.*\)/Copyright 2020 \1, Bernhard Manfred Gruber/'`
`git log --no-merges --after="2021-01-01 00:00" --before="2022-01-01 00:00" --author="Bernhard Manfred Gruber" --name-only --pretty=format:""  | sort -u  | xargs sed -i -e '/Bernhard Manfred Gruber/ s/Copyright [[:digit:]-]\+/Copyright 2021/ ; /Bernhard Manfred Gruber/! s/Copyright [[:digit:]-]\+ \(.*\)/Copyright 2021 \1, Bernhard Manfred Gruber/'`
One manual fix in heatEquation.cpp.
One manual revert of catch2.hpp.
Then clang-format.
`git diff --color-words | sed 's/\x1b\[[0-9;]*m//g' | grep -B 2 202220 | grep  --color=never -o '\(include\|test\)\(.\+\)$' | xargs sed -i -e 's/Copyright [[:digit:]-]\+/Copyright 2022/'`.

j-stephan reviewed (changes requested):
As far as I can see you haven't authored any commits this year. I'm fine with merging this if you remove the year update.

bernhardmgruber commented:
OK, done.

j-stephan commented on example/bufferCopy/src/bufferCopy.cpp:
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2019 Alexander Matthes, Benjamin Worpitz, Erik Zenker, Matthias Werner
+/* Copyright 2019 Alexander Matthes, Benjamin Worpitz, Erik Zenker, Matthias Werner, Bernhard Manfred Gruber
```
Hm, you didn't work for us in 2019. Maybe you can find out the actual date of your commits by using

```
git log --all --after="<date> 00:00" --before="<date> 23:59" --author="<author>"
```

and do this for 2020 and 2021 each? It would be nice to have the year set to "last touched".

bernhardmgruber replied:
done

j-stephan commented on cmake/alpakaCommon.cmake:
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2022 Benjamin Worpitz, Erik Zenker, Axel Huebl, Jan Stephan, René Widera, Jeffrey Kelling, Andrea Bocci
+# Copyright 2021 Benjamin Worpitz, Erik Zenker, Axel Huebl, Jan Stephan, René Widera, Jeffrey Kelling, Andrea Bocci, Bernhard Manfred Gruber
```
This shouldn't happen.

bernhardmgruber replied:
Fixed.

bernhardmgruber commented on include/alpaka/acc/AccCpuFibers.hpp:
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2022 Axel Huebl, Benjamin Worpitz, René Widera, Jan Stephan
+/* Copyright 2021 Axel Huebl, Benjamin Worpitz, René Widera, Jan Stephan, Bernhard Manfred Gruber
```
Ok there are a few year downgrades now.

j-stephan commented:
Since this change is done (mostly) automatically and doesn't add any features to alpaka, would you please set the author to our electronic colleague?

```
git commit --amend --author="Tools <alpaka@hzdr.de>"
```

Commit:
update copyright headers where touched

Commands:
git log --no-merges --after="2020-01-01 00:00" --before="2021-01-01 00:00" --author="Bernhard Manfred Gruber" --name-only --pretty=format:""  | sort -u  | xargs sed -i -e '/Bernhard Manfred Gruber/ s/Copyright [[:digit:]-]\+/Copyright 2020/ ; /Bernhard Manfred Gruber/! s/Copyright [[:digit:]-]\+ \(.*\)/Copyright 2020 \1, Bernhard Manfred Gruber/'
git log --no-merges --after="2021-01-01 00:00" --before="2022-01-01 00:00" --author="Bernhard Manfred Gruber" --name-only --pretty=format:""  | sort -u  | xargs sed -i -e '/Bernhard Manfred Gruber/ s/Copyright [[:digit:]-]\+/Copyright 2021/ ; /Bernhard Manfred Gruber/! s/Copyright [[:digit:]-]\+ \(.*\)/Copyright 2021 \1, Bernhard Manfred Gruber/'
One manual fix in heatEquation.cpp.
One manual revert of catch2.hpp.
Then clang-format.
git diff --color-words | sed 's/\x1b\[[0-9;]*m//g' | grep -B 2 202220 | grep  --color=never -o '\(include\|test\)\(.\+\)$' | xargs sed -i -e 's/Copyright [[:digit:]-]\+/Copyright 2022/'

bernhardmgruber commented:
Changed author to `Tools`.

j-stephan commented on include/alpaka/math/cbrt/CbrtStdLib.hpp:
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2019 Axel Huebl, Benjamin Worpitz
+/* Copyright 2022 Axel Huebl, Benjamin Worpitz, Bernhard Manfred Gruber
```
This doesn't seem right, the file was last touched in 2021.

bernhardmgruber replied:
Seems I accidentially included this while testing the commands. Sorry!

j-stephan commented on test/catch_main/CMakeLists.txt:
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2022 Benjamin Worpitz, Axel Huebl, Jan Stephan
+# Copyright 2021 Benjamin Worpitz, Axel Huebl, Jan Stephan, Bernhard Manfred Gruber
```
There are still some downgrades.

j-stephan commented on script/install_boost.sh:
```diff
@@ -1,7 +1,7 @@
 
 #
-# Copyright 2022 Benjamin Worpitz, René Widera, Axel Huebl, Bernhard Manfred Gruber, Andrea Bocci
+# Copyright 2021 Benjamin Worpitz, René Widera, Axel Huebl, Bernhard Manfred Gruber, Andrea Bocci
```
Here, too.

j-stephan reviewed (changes requested):
Almost there...
